package.json

```diff
@@ -35,6 +35,7 @@
     "clean": "rm -rf node_modules/; rm -rf public/data/QinDenton/",
     "test": "node test/server-test.js && node test/subset-test.js",
     "test-https": "node test/gen_ssl.js && node test/server-test.js --https",
+    "start-https": "node test/gen_ssl.js && node server.js --https",
```

Comment:
This script will generate the SSL certificates and starts the HTTPS server. Detailed instructions are added to the HTTPS section of the README.md file.

Reply:
Do you plan on moving the content of gen_ssl.js into server.js? So the usage would be either

```
node server.js --https
```

or

```
node server.js --https --cert certPath --key keyPath
```

and gen_ssl.js would not be needed.

Reply:
That was the initial idea. Yet, avoided that approach of moving the gen_ssl.js code into server.js since the server should not be triggered until the certificates are completely generated. And only after creating the server, we should be calling the [server.listen.](https://github.com/hapi-server/server-nodejs/blob/f42ec2f5ba2b6c3a144121dee8b67a379ee78f2f/server.js#L276) Since [https.createServer](https://github.com/hapi-server/server-nodejs/blob/f42ec2f5ba2b6c3a144121dee8b67a379ee78f2f/server.js#L136) takes the path of SSL certificates in the form of [options](https://github.com/hapi-server/server-nodejs/blob/f42ec2f5ba2b6c3a144121dee8b67a379ee78f2f/server.js#L132), we would be ending up having many callbacks for generating the [certificates](https://github.com/hapi-server/server-nodejs/blob/d06ac5ef0ee0a9e1718688696f5b989aec4ce7a6/test/gen_ssl.js#L6) , then [ creating the server](https://github.com/hapi-server/server-nodejs/blob/f42ec2f5ba2b6c3a144121dee8b67a379ee78f2f/server.js#L136), and [listening](https://github.com/hapi-server/server-nodejs/blob/f42ec2f5ba2b6c3a144121dee8b67a379ee78f2f/server.js#L276) to the specified port. All these callbacks would make the server.js messy.

Reply:
I'm guessing that execSync won't work in windows so as to prevent the need for a callback. Is that correct? This would remove the need for a callback.

Does gen.sh work in Windows too?

If we are to keep things separate, I don't know that gen_ssl.js is needed as its only job is to call gen.sh.

Reply:
Yes, gen.sh works in windows too. As you mentioned, execSync removes the need for a callback when generating certificates. I will get rid of gen_ssl.js and merge it into server.js.